### DISCUSSION TITLE
Internal visibility for personal forks of enterprise repositories with internal visibility

### DISCUSSION THREAD

**Discussion by mihaic** [community/community#57485] [Enterprise]
### Select Topic Area

Product Feedback

### Body

I find it useful to collaborate on pull requests from personal forks using the "[Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)" GitHub feature. However, the default private visibility of personal forks of enterprise repositories with [internal visibility](https://docs.github.com/en/enterprise-cloud@latest/repositories/creating-and-managing-repositories/about-repositories#about-internal-repositories) prevents this collaboration. Please consider allowing such personal forks to have internal visibility.

I know that organization-owned forks have internal visibility, but the "Allow edits by maintainers" feature only works for personal forks.

I currently use personal forks and manually give permission to upstream maintainers, but that is cumbersome. Is there another solution I am missing to enable internal collaboration on PRs?

**Comment by msgongora** [comment]
I also second this request. It is a contradiction that you can't access a fork as an upstream collaborator if the PR's owner has selected "[Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)". It won't allow the collaborator to edit the code. It will only allow to fix conflicts and sync the PR branch with the destination branch through the GitHub web UI.  I manage/support multiple enterprise organizations and here is what I have learned during my years doing so and using the forking workflow:

1. The way that Github has implemented it today, forking doesn't compromise security.
2. It makes management easier from the access point of view. You need to collaborate on a repo from a Github Org within your enterprise that you are not collaborator? Not a problem... just fork and file a PR. No need to bother a repo admin or a Github Org owner to include you.
3. It is cleaner for CI/CD. You don't need to worry about cleaning repositories with hundreds of branches left behind by developers in repos where automatic ref deletion can't be enabled because of some development workflow requirement. 

I also think that besides this feature request, GitHub organization owners and repo admins should be able to enforce the "[Allow edits by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)". That way, any PR opened from a fork will allow collaborators from upstream to edit the PR. While leaving the authority to enable/disable this option to the PR's author, it is nice for public/open-source repos, enterprise doesn't need to follow the same rule.
